.travis.yml

```diff
@@ -1,29 +1,50 @@
 language: python
 sudo: false
-python:
-    - 2.7
-    - 3.4
-    - 3.5
-    - 3.6
-env:
-    - ENVIRON=py
-    - ENVIRON=py27-rp3,py27-datetime,py36-datetime
-    - ENVIRON=isort,flake8,docs
+
 matrix:
-    exclude:
-        - env: ENVIRON=isort,flake8,docs
-        - env: ENVIRON=py27-rp3,py27-datetime,py36-datetime
     include:
+        - python: "2.7"
+          env: TOXENV=docs,lint-py2
         - python: "3.6"
-          env: ENVIRON=py36-datetime,isort,flake8,docs
+          env: TOXENV=docs,lint-py3
+        - python: "2.7"
+          env: TOXENV=py27
+        - python: "2.7"
+          env: TOXENV=py27-rp3
         - python: "2.7"
-          env: ENVIRON=py27-rp3,py27-datetime
+          env: TOXENV=py27-datetime
+        - python: "3.4"
+          env: TOXENV=py34
+        - python: "3.5"
+          env: TOXENV=py35
+        - python: "3.6"
+          env: TOXENV=py36
+        - python: "3.6"
+          env: TOXENV=py36-datetime
+        - python: "3.7-dev"
+          env: TOXENV=py37
+        - python: "pypy"
+          env: TOXENV=pypy
+        - python: "pypy3"
+          env: TOXENV=pypy
+    allow_failures:
+        - python: "3.7-dev"
+          env: TOXENV=py37
+        - python: "pypy"
+          env: TOXENV=pypy
+        - python: "pypy3"
+          env: TOXENV=pypy
+
 install:
-    - pip install tox coveralls coverage
+    - travis_retry pip install -U pip setuptools
+    - travis_retry pip install -U -c constraints.txt tox coveralls coverage
```

Comment:
`-c constraints.txt` is a no-op as the file is empty now, so it should be removed.

Reply:
see comment for the constraints.txt